index.js

```diff
@@ -139,12 +142,65 @@ module.exports = {
     return fastbootBuild.toTree();
   },
 
+  serverMiddleware: function(options) {
+    var emberCliVersion = this._getEmberCliVersion();
+    var app = options.app;
+    var options = options.options;
+
+    if (emberCliVersion.satisfies('>= 2.12.0-beta.1')) {
+      // only run the middleware when ember-cli version for app is above 2.12.0-beta.1 since
+      // that version contains API to hook fastboot into ember-cli
+
+      app.use(function(req, resp, next) {
+        var broccoliHeader = req.headers['x-broccoli'];
+        var outputPath = broccoliHeader['outputPath'];
+
+        if (broccoliHeader['url'] === req.serveUrl) {
+          // if it is a base page request, then have fastboot serve the base page
+          if (!this.fastboot) {
+            // TODO: make this configurable for allowing apps to pass sandboxGlobals
+            // and custom sandbox class
+            this.ui.writeLine(chalk.green('App is being served by FastBoot'));
+            this.fastboot = new FastBoot({
+              distPath: outputPath
+            });
+          }
+
+          var fastbootMiddleware = FastBootExpressMiddleware({
+            fastboot: this.fastboot
+          });
+
+          fastbootMiddleware(req, resp, next);
+        } else {
+          // forward the request to the next middleware (example other assets, proxy etc)
+          next();
+        }
+      }.bind(this))
+    }
+  },
+
   outputReady: function() {
     this.emit('outputReady');
   },
 
-  postBuild: function() {
+  postBuild: function(result) {
     this.emit('postBuild');
+    if (this.fastboot) {
+      // should we reload fastboot if there are only css changes? Seems it maynot be needed.
+      // TODO: we can do a smarter reload here by running fs-tree-diff on files loaded
+      // in sandbox (explore later).
+      this.ui.writeLine('Reloading FastBoot...');
+      this.fastboot.reload({
```

Comment:
We can implement the `instrumentation` API (ember-cli@2.13+) and detect the output files that have changed, to see if we need to reload `fastboot` here.